### PR TITLE
This is a merge of our current project

### PR DESCRIPTION
**Pull request by ostlerc**
we have a hello world wxwidgets with the catch framework and using easylog for (thread safe) logging.

### CODE REVIEW

**Comment by ostlerc** [comment]
this doesn't need to be merged, but it is to show the progress we are making and show early our direction in case we have misunderstood something

**Comment by tehul** [comment]
I got an error when I tried running make on the project.
```
% make
Scanning dependencies of target common
[ 14%] Building CXX object CMakeFiles/common.dir/common/sample.cpp.o
[ 28%] Building CXX object CMakeFiles/common.dir/common/log/logger.cpp.o
In file included from /home/trevor/dev/audit/CS5500/common/log/logger.h:6:0,
                 from /home/trevor/dev/audit/CS5500/common/log/logger.cpp:1:
/home/trevor/dev/audit/CS5500/common/log/easylogging++.h:1308:35: error: unused parameter ‘ms’ [-Werror=unused-parameter]
     static inline void msleep(int ms)
                                   ^
cc1plus: all warnings being treated as errors
CMakeFiles/common.dir/build.make:77: recipe for target 'CMakeFiles/common.dir/common/log/logger.cpp.o' failed
make[2]: *** [CMakeFiles/common.dir/common/log/logger.cpp.o] Error 1
CMakeFiles/Makefile2:60: recipe for target 'CMakeFiles/common.dir/all' failed
make[1]: *** [CMakeFiles/common.dir/all] Error 2
Makefile:86: recipe for target 'all' failed
```
Github won't let me comment on `easylogging++.h` because it's too big, but I changed `msleep`'s non-Windows implementation to the Window's implementation to get it to compile. 
```c++
    static inline void msleep(int)
    {
    }
```
I'm not sure if this is a good way to handle this. I also tried adding `#define ELPP_EXPERIMENTAL_ASYNC` to `logger.h`, but that just caused `dwfort` to segfault. Any ideas?

**Comment by ostlerc** [comment]
I think removing the parameter is fine as long as we never have ELPP_ASYNC_LOGGING (ELPP_EXPERIMENTAL_ASYNC) enabled.

**Comment by ostlerc** [comment]
I have pushed the file with no variable.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,24 +1,68 @@
 cmake_minimum_required(VERSION 2.8)
-project(TorusWorld) # Very tentative project name.
+project(dwfort)
```
Comment (ksundberg) [inline CMakeLists.txt]: I understand that dwarf fortress is a source of inspiration, but we need a different name for our project before I can accept this commit.

File: client/forms/MainWindow.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#ifndef _MAINWINDOW_H_
+#define _MAINWINDOW_H_
+
+#include "wxwidget.h"
```
Comment (ksundberg) [inline client/forms/MainWindow.h]: wx is a third party library and should be included with <> not " "

File: common/noise/PerlinNoise.cpp [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+  return (omega(v[0]) * omega(v[1]) * omega(v[2]) *
+          (this->dot(gamma(i, j, k), v)));
+}
+
+double PerlinNoise::dot(vector3d left, vector3d right) const
+{
+  return left[0] * right[0] + left[1] * right[1] + left[2] * right[2];
+}
+
+double PerlinNoise::turbulence2D(double x, double y, int depth) const
```
Comment (ksundberg) [inline common/noise/PerlinNoise.cpp]: There was are request for a turbulence1D function also.

File: CMakeLists.txt (hunk 1 above)
Comment (ostlerc) [inline CMakeLists.txt]: Agreed

**Pull request merged**